### PR TITLE
Issue #2952: Move core layout contexts into their respective modules.

### PR DESCRIPTION
**Pull request by jenlampton**
Fixes https://github.com/backdrop/backdrop-issues/issues/2952
Replaces https://github.com/backdrop/backdrop/pull/2080

### CODE REVIEW

File: core/modules/user/user.module [hunk 1]
```diff
@@ -136,6 +136,23 @@ function user_entity_info() {
   );
 }
 
+/**
+ * Implements hook_layout_context_info().
+ */
+function user_layout_context_info() {
+  $info['user'] = array(
+    'title' => t('User'),
```
Comment (opi) [inline core/modules/user/user.module]: Shouldn't we use "User account" here ? ( Source: https://github.com/backdrop/backdrop-issues/issues/127 )

File: core/modules/layout/includes/layout.layout.inc [hunk 2]
```diff
@@ -25,26 +25,6 @@ function layout_layout_context_info() {
     // main context data.
     'load callback' => 'node_load',
   );
```
Comment (opi) [inline core/modules/layout/includes/layout.layout.inc]: What about moving this *node* layout context into `node.module` too ?

**Review by opi — COMMENTED** [review]
LGTM

File: core/modules/user/user.module (hunk 1 above)
Comment (jenlampton) [reply core/modules/user/user.module]: Yes! nice catch :)

File: core/modules/layout/includes/layout.layout.inc (hunk 2 above)
Comment (jenlampton) [reply core/modules/layout/includes/layout.layout.inc]: I wondered about that. I'll give it a shot :)

File: core/modules/user/user.module (hunk 1 above)
Comment (jenlampton) [reply core/modules/user/user.module]: fixed.

File: core/modules/layout/includes/layout.layout.inc (hunk 2 above)
Comment (jenlampton) [reply core/modules/layout/includes/layout.layout.inc]: done!

**Review by opi — APPROVED** [review]

**Comment by quicksketch** [comment]
Thanks, merged into 1.x and 1.9.x.

**Pull request merged**